tests/integration/renderer/test_loading_states.py

```diff
@@ -167,3 +169,46 @@ def find_text(spec):
 
     find_spinners()
     find_text({1: 1, 2: 1, 3: 1, 4: 1})
+
+
+@pytest.mark.parametrize(
+    "kwargs, expected_update_title",
+    [
+        ({}, "Updating..."),
+        ({"update_title": None}, "Dash"),
+        ({"update_title": ""}, "Dash"),
+        ({"update_title": "Hello World"}, "Hello World"),
+    ]
+)
+def test_rdls003_update_title(dash_duo, kwargs, expected_update_title):
+    app = dash.Dash("Dash", **kwargs)
+    lock = Lock()
+
+    app.layout = html.Div(
+        children=[
+            html.H3("Press button see document title updating"),
+            html.Div(id="output"),
+            html.Button("Update", id="button", n_clicks=0),
+        ]
+    )
+
+    @app.callback(
+        Output("output", "children"),
+        [Input("button", "n_clicks")]
+    )
+    def update(n):
+        with lock:
+            # check for update-title while processing callback
+            until(lambda: dash_duo.driver.title == expected_update_title, timeout=1)
```

Comment:
Interesting... this appears to work but I'm kind of surprised it does - it might actually be waiting for the timeout period and then throwing an error silently. That's because the server and the browser side are running in different processes (hence the need for `Lock`).

I'd remove the title check inside the callback, and just put it at the very end of the test, after `#button.click()` (but still inside the `with lock:` block)

Reply:
@alexcjohnson 
> I'd remove the title check inside the callback, and just put it at the very end of the test, after #button.click() (but still inside the with lock: block)

This was my first shot but that didn't work at that time because I missed the lock inside the callback. Now this works like a charm. I also added the check on startup again.